synapse/util/caches/lrucache.py

```diff
@@ -142,10 +140,11 @@ def cache_clear():
             list_root[NEXT] = list_root
             list_root[PREV] = list_root
             cache.clear()
+            self.size = 0
```

Comment:
I dislike the `LruCache` trying to keep track of the size of the underlying cache. I much prefer that caches/collections keep track of their own size.